java-checks/src/main/java/org/sonar/java/checks/LambdaSingleExpressionCheck.java

```diff
@@ -74,4 +82,67 @@ private static boolean singleStatementIsReturn(LambdaExpressionTree lambdaExpres
   private static boolean isReturnStatement(Tree tree) {
     return tree.is(Tree.Kind.RETURN_STATEMENT);
   }
+
+  record InvocationData(Symbol.MethodSymbol methodSymbol, int argumentPosition) {}
+
+  /**
+   * Assuming {@code tree} is an argument in a method invocation,
+   * get the symbol of the method and the position of the argument.
+   */
+  private static Optional<InvocationData> getInvocationData(Tree argument) {
+    Optional<Arguments> argumentList = Optional.ofNullable(argument.parent())
+      .filter(Arguments.class::isInstance)
+      .map(Arguments.class::cast);
+
+    if(argumentList.isPresent()) {
```

Comment:
I would prefer `if (argument.parent() instance of Arguments argumentList)` which is more succinct